src/methods/geom_relations/geom_geom_processors.jl

```diff
@@ -0,0 +1,739 @@
+
+If the point is in an "allowed" location, return true. Else, return false.
+=#
+function _point_polygon_process(
+    point, polygon;
+    in_allow, on_allow, out_allow,
+)
+    # Check interaction of geom with polygon's exterior boundary
+    ext_val = _point_filled_curve_orientation(point, GI.getexterior(polygon))
+    # If a point is outside, it isn't interacting with any holes
+    ext_val == point_out && return out_allow
+    # if a point is on an external boundary, it isn't interacting with any holes
+    ext_val == point_on && return on_allow
+    
+    # If geom is within the polygon, need to check interactions with holes
+    for hole in GI.gethole(polygon)
+        hole_val = _point_filled_curve_orientation(point, hole)
+        # If a point in in a hole, it is outside of the polygon
+        hole_val == point_in && return out_allow
+        # If a point in on a hole edge, it is on the edge of the polygon
+        hole_val == point_on && return on_allow
+    end
+    
+    # Point is within external boundary and on in/on any holes
+    return in_allow
+end
+
+#=
+Determines if a line meets the given checks with respect to a curve.
+
+If over_allow is true, segments of the line and curve can be co-linear.
+If cross_allow is true, segments of the line and curve can cross.
+If on_allow is true, endpoints of either the line or curve can intersect a 
+    segment of the other geometry.
+If cross_allow is true, segments of the line and curve can be disjoint.
+
+If in_require is true, the interiors of the line and curve must meet in at least
+    one point.
+If on_require is true, the bounday of one of the two geometries can meet the
+    interior or boundary of the other geometry in at least one point.
+If out_require is true, there must be at least one point of the given line that
+    is exterior of the curve.
+
+If the point is in an "allowed" location and meets all requirments, return true.
+Else, return false.
+
+If closed_line is true, line is treated as a closed line where the first and
+last point are connected by a segment. Same with closed_curve.
+=#
+function _line_curve_process(
+    line, curve;
+    over_allow, cross_allow, on_allow, out_allow,
+    in_require, on_require, out_require,
+    closed_line = false,
+    closed_curve = false,
+)
+    # Set up requirments
+    in_req_met = !in_require
+    on_req_met = !on_require
+    out_req_met = !out_require
+    # Determine curve endpoints
+    nl = GI.npoint(line)
+    nc = GI.npoint(curve)
+    first_last_equal_line = equals(GI.getpoint(line, 1), GI.getpoint(line, nl))
+    first_last_equal_curve = equals(GI.getpoint(curve, 1), GI.getpoint(curve, nc))
+    nl -= first_last_equal_line ? 1 : 0
+    nc -= first_last_equal_curve ? 1 : 0
+    closed_line |= first_last_equal_line
+    closed_curve |= first_last_equal_curve
+    # Loop over each line segment
+    l_start = GI.getpoint(line, closed_line ? nl : 1)
+    i = closed_line ? 1 : 2
+    while i ≤ nl
```

Comment:
This is an epic `while` loop... some vertical compression could help keeping it in your head

(also btw you can use 92 chars width in julia, may help a little)